src/ripple/json/MultivarJson.h

```diff
@@ -0,0 +1,109 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2023 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_BASICS_MULTIVARJSON_H_INCLUDED
+#define RIPPLE_BASICS_MULTIVARJSON_H_INCLUDED
+
+#include <ripple/json/json_value.h>
+
+#include <array>
+#include <concepts>
+#include <cstdlib>
+
+namespace ripple {
+template <std::size_t Size>
+struct MultivarJson
+{
+    std::array<Json::Value, Size> val;
+    constexpr static std::size_t size = Size;
+
+    Json::Value const&
+    select(auto&& selector) const
```

Comment:
For `select`: do not want to implicitly invoke any of the builtin conversion to `std::size_t` when the user lambda returns e.g. an `int`. IMO that's rather important because `std::size_t` is unsigned.
For `set`: do not want compilation error when Json element cannot be assigned from the parameter; would rather drop `set` from the overload set instead. Not that important, but it's SFINAE-friendly design

Reply:
I see, what about:
```
    Json::Value const&
    select(auto selector) const requires std::same_as(decltype(s), std::size_t) // no implicit conversion
    {
        return val[selector];
    }
    // would we still need a callable for selector?
```

Regarding your concern about the `set` method, all `Json::Value`'s should be assignable right? In what cases would we get a compilation error?

Reply:
Re. design choice in `select` - passing a functor (or lambda) allows for improved code cohesion. I make use of it in `apiVersionSelector` - as you will notice, this makes coupling between index and api version more explicit and limited to one file. Well, in principle - I could make a better use of it.

Re. `set` , the error I meant was from assignment to `Json::Value` from some type which cannot be assigned to it, like for example demonstrated in unit tests `std::nullopt`

Reply:
Hmm alright, ok